packages/@webex/plugin-meetings/src/statsAnalyzer/index.ts

```diff
@@ -598,203 +552,219 @@ export class StatsAnalyzer extends EventsScope {
             );
           }
 
           if (
-            this.statsResults.resolutions[mediaType].send.framesSent ===
-              this.lastStatsResults.resolutions[mediaType].send.framesSent ||
-            this.statsResults.resolutions[mediaType].send.framesSent === 0
+            this.statsResults.resolutions['video-share-send'].send.framesSent ===
+              this.lastStatsResults.resolutions['video-share-send'].send.framesSent ||
+            this.statsResults.resolutions['video-share-send'].send.framesSent === 0
           ) {
             LoggerProxy.logger.info(
-              `StatsAnalyzer:index#compareLastStatsResult --> No ${mediaType} frames sent`
+              `StatsAnalyzer:index#compareLastStatsResult --> No share frames sent`
             );
           }
         }
+      }
 
+      if (this.meetingMediaStatus.expected.sendShare) {
         // TODO:need to check receive share value
-        // compare share stats reveived
-        currentStats = this.statsResults[mediaType].recv;
-        previousStats = this.lastStatsResults[mediaType].recv;
+        // compare share stats received
+        const currentPacketsReceived = getCurrentStatsTotals(
+          'video-share-recv',
+          'totalPacketsReceived'
+        );
+        const currentPacketsSent = getCurrentStatsTotals('video-share-recv', 'totalPacketsSent');
+        const previousPacketsReceived = getPreviousStatsTotals(
+          'video-share-recv',
+          'totalPacketsReceived'
+        );
+        const currentFramesReceived = getCurrentResolutionsStatsTotals(
+          'video-share-recv',
+          'framesReceived'
+        );
+        const previousFramesReceived = getPreviousResolutionsStatsTotals(
+          'video-share-recv',
+          'framesReceived'
+        );
+        const currentFramesDecoded = getCurrentStatsTotals('video-share-recv', 'framesDecoded');
+        const previousFramesDecoded = getPreviousStatsTotals('video-share-recv', 'framesDecoded');
+        const currentFramesDropped = getCurrentResolutionsStatsTotals(
+          'video-share-recv',
+          'framesDropped'
+        );
+        const previousFramesDropped = getPreviousResolutionsStatsTotals(
+          'video-share-recv',
+          'framesDropped'
+        );
 
-        if (
-          currentStats.totalPacketsReceived === previousStats.totalPacketsReceived ||
-          currentStats.totalPacketsSent === 0
-        ) {
+        const senderFramesDecoded = Object.keys(this.statsResults)
+          .filter((keys) => keys.includes('video-share-recv'))
+          .reduce(
+            (prev, cur) =>
+              prev +
+              ((this.statsResults.resolutions[cur].send &&
+                this.statsResults.resolutions[cur].send.framesDecoded) ||
+                0),
+            0
+          );
+
+        if (currentPacketsReceived === previousPacketsReceived || currentPacketsSent === 0) {
```

Comment:
this doesn't look right - "currentPacketsSent" doesn't belong here, should be `currentPacketsReceived` right?

Reply:
Correct, updated.